name: Status check functions
on: push
jobs: 
  fail: 
    runs-on: ubuntu-latest
    steps:
      - run: exit 1 #終了ステータスが0以外なので、エラーが発生する
      - run: echo "Hello"
        if: ${{ failure() }} #エラーが発生しても、failure関数があるのでHelloの出力は実行される
        # failure関数は手前の処理が失敗したらtrue
